pippy/PipelineStage.py

```diff
@@ -421,6 +421,55 @@ def split_inputs(
             # Return a list of empty tuples/dicts with matching length as chunks
             return [()] * self.chunks, [{}] * self.chunks
 
+    def get_fwd_recv_ops(self) -> List[dist.P2POp]:
+        """
+        Returns a list of ops that are needed to receive the input arguments
+        for this stage.
+        """
+        ops = []
+        recv_infos = self.args_recv_info[self.fwd_chunk_id]
+        for info in recv_infos:
+            if isinstance(info, RecvInfo):
+                peer_rank = self.stage_index_to_group_rank[info.source]
+                peer_global_rank = (
+                    peer_rank
+                    if self.group is None
+                    else dist.get_global_rank(self.group, peer_rank)
+                )  # TODO
```

Comment:
What is the TODO for?

Reply:
It is for c10d I believe. 
```
dist.get_global_rank(self.group, peer_rank)
```
seems to error out when `self.group = None`.